doc/developer/design/20220322_multi_table_transactions.md

```diff
@@ -0,0 +1,48 @@
+# Multi-table Transactions
+
+## Summary
+
+Upcoming persist changes will remove the persist API allowing writing to multiple collections in an atomic operation.
+In order to allow the persist API to be written, we must do one of:
+
+1. Disallow multi-table transactions.
+2. Use an existing (or upcoming) materialize API to record transactions.
+3. Use a third-party product to record transactions.
+
+We need a log to record transactions that can write the entire transaction atomically, which will allow the rest of the system to know that the transaction committed.
```

Comment:
note: comment below points to a difference in our understanding here.

That  the "transaction committed" is a little ambiguous here; it's more like "the client tried to commit the transaction" or the "transaction successfully committed to the WAL." In the saga-style design (where storage nodes attempt to apply items in the WAL), the transactions' writes are not committed until the storage nodes in plurality deem it acceptable. I think this is what you're saying, just a nudge that my understanding would be enhanced if this were less ambiguous.